### ISSUE TITLE
Add-on Store: When updating an incompatible add-on, the updated add-on will disappear from the list.

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#16015]
### Steps to reproduce:
Try updating an add-on in the list of updateable add-ons.
### Actual behavior:
Click "Update" and the add-on disappears from the list.
This prevents users from viewing the download status of the add-on.
### Expected behavior:
Add-ons should not disappear from the list allowing users to view their download status.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2024.1Beta2, 2024.2Alpha
#### Windows version:
Windows 10 22H2 (AMD64) build 19045.3803
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
not related

**Comment by Adriani90** [comment]
I don't really understand why this separate tab is needed, it only introduces inconsistencies and is inefficient. There should really be a way to make updatable addons visible to the user in another way (i.e. via Windows toast notifications or as a mechanism in the installed addons list. The current approach is not really consistent with other usual software design, I don't know any software with addons that places updatable addons in a list separated from the installed addons list. I really try to understand the advantages from the NVDA addon store updatable addons tab approach but i really struggle to understand it. However, these are only my 5 cents on this.

**Comment by Brian1Gaff** [comment]
Is this not so that the can remain  for a lot longer than toasts etc do?
 Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
  ----- Original Message -----

**Comment by cary-rowen** [comment]
Seriously, I don’t fully understand what @Adriani90 and @Brian1Gaff are talking about.
@Adriani90 is probably saying that there shouldn't be an "updatable add-ons" tab, but instead notify the user that an add-on is updateable, and integrate it into the list of installed add-ons.

I don't think the current implementation is that bad:
* First, notifications are easier to ignore.
* Second, it doesn’t seem that convenient for users to manually check which add-ons have new versions in the list.
In the worst case, the user needs to browse through each list item completely until they hear that the add-on is updatable.
In a slightly better situation, the user might know to use table navigation and browse the status column individually.
In this case the user still needs to traverse the entire list.

I'm really sorry @Brian1Gaff I don't understand what you are talking about at all. If your comment has nothing to do with this Issue, I suggest you hide or delete it.
@Adriani90 No matter how much you hate the current situation, I also suggest you open a new Issue or discussion to discuss your views, thanks!

**Comment by CyrilleB79** [comment]
Reproduced with 2024.1beta3 but not with 2023.3 on my side.

@seanbudd, since it is a regression introduced in 2024.1 dev cycle, I'd strongly recommend to add this issue in 2024.1 milestone. Introducing new bugs in the Add-on Store while resolving other ones gives a very bad impression of this feature,
more especially given the Add-on Store has not yet fully adopted by the community and its development is not seen as fully completed (e.g. #14986  and duplicates, #3208, #15035, etc. and many comments on the lists indicating that Add-on Updater or Tienda add-ons are still widely used by people, including myself).

**Comment by CyrilleB79** [comment]
@Adriani90 in answer to your comment:

> I don't really understand why this separate tab is needed, it only introduces inconsistencies and is inefficient.

You are asking the deletion of some or all tabs in various issues. Since this request is scattered among many issues, it's very hard to follow and inefficient. If your goal is to remove tabs in the Add-on Store, it would be clearer to describe and summarize in a new separate issue the UX that you suggest in a specific issue, highlighting the pros and cons of your new design with concrete use case examples, or in a discussion.

The current issue describes a very specific issue in the current design; please do not broaden the topic with a full GUI redesign request here. I guess (and hope) that the current issue can be solved in beta branch whereas I am sure that, if any, a new GUI design would not be introduced in the beta branch.

> There should really be a way to make updatable addons visible to the user in another way (i.e. via Windows toast notifications 

Toast notifications are not incompatible with the current design, and IMO, it should have been provided along with the Store since its introduction.

> The current approach is not really consistent with other usual software design, I don't know any software with addons that places updatable addons in a list separated from the installed addons list.

Providing concrete example describing how it is done in other softwares would really be helpful indeed to progress this topic; if you open a new summary issue, please provide such examples there.

**Comment by CyrilleB79** [comment]
In addition when clicking "Update" on an add-on, the list is updated again, probably requesting again information from the add-on store server. On some machine with a slow or unstable connection, this updates may take a few seconds, which is very bad experience. Indeed, when I click "Update", I expect to be able to browse immediately the rest of the list while the add-on downloads in the background, which is not possible now.

@seanbudd, let me know if it's considered a separate issue or not.

**Comment by seanbudd** [comment]
@CyrilleB79 - new data shouldn't be fetched unless the cache hash changes. As far as I am aware, this is just a re-draw of the view models. I think it's necessary to update deeper parts of the view models such as actions, but we can probably split this out to keep the add-on lists static and only provide the deeper refresh.
I think this should be raised as a separate issue
